### ISSUE TITLE
cannot set allowIncompleteChunks to false

### ISSUE THREAD

**Issue by gangseok514** [operator-framework/operator-metering#922]
Hello, I am looking into operater-metering sourcecode recently due to some issues.

I set pollInterval: 10m, chunksize: 10m, stepsize: 5m

In the importer.go and query.go sourcecode,
lastTimestamp is set current time because of allowIncompleteChunks,
so the next collected metric is unintentional.

example:
```
time="2019-09-05T10:19:31Z" level=info msg="stored a total of 700 metrics for data between 2019-09-05 10:17:00 +0000 UTC and 2019-09-05 10:19:00 +0000 UTC into datasource_gpu_request_slots" ReportDataSource=gpu-request-slots app=metering chunkSize=10m0s component=PrometheusImporter endTime="2019-09-05 10:19:00.044390211 +0000 UTC" logID=5cT8O1v0rR queryName=gpu-request-slots reportDataSource=gpu-request-slots startTime="2019-09-05 10:17:00 +0000 UTC" stepSize=5m0s tableName=datasource_gpu_request_slots

I0905 19:31:58.603887   20495 kubectl.go:56] Execution time : 463.397807ms
time="2019-09-05T10:29:37Z" level=info msg="stored a total of 1431 metrics for data between 2019-09-05 10:24:00 +0000 UTC and 2019-09-05 10:29:00 +0000 UTC into datasource_gpu_request_slots" ReportDataSource=gpu-request-slots app=metering chunkSize=10m0s component=PrometheusImporter endTime="2019-09-05 10:29:33.589955865 +0000 UTC" logID=5cT8O1v0rR queryName=gpu-request-slots reportDataSource=gpu-request-slots startTime="2019-09-05 10:24:00 +0000 UTC" stepSize=5m0s tableName=datasource_gpu_request_slots
```

In fact, I would like to collect metric at 10:17 , 10:22(+5m), 10:27(+5m) by stepsize.

but lastTimestamp is set 10:19 because of allowIncompleteChunks,
So the unintentional metrics between 10:24(10:19+5m) and 10:29 are collected.

```golang
func getTimeRangesChunked(beginTime, endTime time.Time, chunkSize, stepSize time.Duration, maxTimeRanges int64, allowIncompleteChunks bool) []prom.Range {
	chunkStart := truncateToSecond(beginTime)
	chunkEnd := truncateToSecond(chunkStart.Add(chunkSize))

	// don't set a limit if negative or zero
	disableMax := maxTimeRanges <= 0

	var timeRanges []prom.Range
	for i := int64(0); disableMax || (i < maxTimeRanges); i++ {
		if allowIncompleteChunks {
			if chunkEnd.After(endTime) {
				chunkEnd = truncateToSecond(endTime)
			}
			if chunkEnd.Equal(chunkStart) {
				break
			}
		} else {
			// Do not collect data after endTime
			if chunkEnd.After(endTime) {
				break
			}

			// Only get chunks that are a full chunk size
			if chunkEnd.Sub(chunkStart) < chunkSize {
				break
			}
		}
```

If I set allowIncompleteChunks to false, this problem will be fixed.
But allowIncompleteChunks is hard-coded.

```golang
const (
	reportDataSourceFinalizer = cbTypes.GroupName + "/reportdatasource"
	partitionUpdateInterval   = 30 * time.Minute
	// allowIncompleteChunks must be true generally if we have a large
	// chunkSize because otherwise we will wait for an entire chunks worth of
	// data before importing metrics into Presto.
	allowIncompleteChunks = true
)
```

Can I know why this variable is not created as option?
and Is it okay if I change this to false, or make this optional?

Thank you.

**Comment by chancez** [comment]
I'm having trouble understanding the actual issue your trying to report. Is it missing time periods, based on your logs it looks like `10:19:00` to `10:24:00` is missing? Or are you saying the problem is that it's querying chunks smaller than the `stepSize`?

**Comment by gangseok514** [comment]
first import runs at 10:19. it collects metric between 10:17~19. it's timestamp which is stored into db is 10:17.

next runs at 10:29.
it collects metric between 10:24~29. and it's timestamp is 10:24 and 10:29

db timestamp
- 10:17
- 10:24
- 10:29

but it is not the result i expected.
i set stepsize 5m. the result i expect is the below.
- 10:17
- 10:22
- 10:27

the reason for the difference is because of allowIncompleteChunks.

**Comment by chancez** [comment]
I see the issue, I'm trying to decide whether we should even have `allowIncompleteChunks` at all at this point or if I can just remove it.

**Comment by chancez** [comment]
Fixed by #925

**Comment by gangseok514** [comment]
I am sad that I missed a good opportunity to contribute... 😢
